docs/guides/upgrading-to-standard-linter-v2.md

```diff
@@ -0,0 +1,86 @@
+# Upgrading to Standard Linter v2
+
+This upgrade guide assumes your package uses Standard Linter API v1 and Message API v1.
+
+## package.json
+
+You need to update the version in the manifest for `linter` service from `1.x.x` to `2.0.0`.
+
+**Before:**
+
+```
+... stuff above
+  "providedServices": {
+    "linter": {
+      "versions": {
+        "1.0.0": "provideLinter"
+      }
+    }
+  },
+... stuff below
+```
+
+**After:**
+
+```
+... stuff above
+  "providedServices": {
+    "linter": {
+      "versions": {
+        "2.0.0": "provideLinter"
+      }
+    }
+  },
+... stuff below
+```
+
+## index.js
+
+In the newer version, the `name` property is now mandatory. Providers must also return Message v2 instead of Message v1.
```

Comment:
Maybe something more along these lines:

```markdown
The new Message format is quite different from the v1 format, although all the essential elements are still there. The changes in required information are:

* `type` is no longer allowed and...
* `severity` is now mandatory
* `text` is renamed to `excerpt`
* `html` is no longer allowed, similar functionality is available as `description` which is either a string or a function (Promise) returning a string which is then rendered as Markdown.
* `name` is no longer allowed
* `filePath` and `range` have been combined into a mandatory `location` property
* `trace` no longer exists, the functionality has been split into `source` (for the `location` equivalent) and `reference` for the `text` equivalent.
* `fix` has been replaced with `solutions`, which is far more flexible.
* `selected` is no longer allowed
```

Hopefully I'm accurate in the descriptions there, if not fix it and I'll see how it looks in the next round of review 😉.